middle_end/flambda/terms/cost_metrics.rec.ml

```diff
@@ -119,15 +129,37 @@ let (+) a b ={
 (* The metrics for a set of closures are the sum of the metrics for each closure
    it contains. The intuition behind it is that if we do inline a function f in
    which a set of closure is defined then we will copy the body of all functions
-   referred by this set of closure as they are dependent upon f.*)
-let set_of_closures ~find_cost_metrics set_of_closures =
+   referred by this set of closure as they are dependent upon f.
+
+   A set of closures introduces implicitly an alloc whose size (as in OCaml 4.11)
+   is:
+     total number of closure variables + sum of s(arity) for each closure
+   where s(a) = if a = 1 then 2 else 3
+*)
+let set_of_closures ~find_cost_characteristics set_of_closures =
   let func_decls = Set_of_closures.function_decls set_of_closures in
   let funs = Function_declarations.funs func_decls in
-  Closure_id.Map.fold (fun _ func_decl metrics ->
+  let num_clos_vars =
+    Set_of_closures.closure_elements set_of_closures
+    |> Var_within_closure.Map.cardinal
+  in
+  let cost_metrics, num_blocks =
+    Closure_id.Map.fold (fun _ func_decl (metrics, num_blocks) ->
       let code_id = Function_declaration.code_id func_decl in
-      metrics + find_cost_metrics code_id
+      let code_metrics, code_arity = find_cost_characteristics code_id in
+      metrics + code_metrics,
+      (* CR poechsel: valid until OCaml 4.13 *)
```

Comment:
Maybe also note in this CR that the code above in `named_size` will also need fixing for 4.13.